.github/workflows/main.yml

```diff
@@ -32,3 +32,6 @@ jobs:
 
       - name: Check Types
         run: yarn lint:types
+
+      - name: Run Unit Tests
+        run: yarn test:unit
```

Comment:
probably we want to run this in a matrix, and not in the same job as linting/typechecking

Reply:
Matrix seemed like a overkill for PRs so moved for separate check on master btw.

Reply:
I strongly disagree it’s overkill; it’s very common for a PR to work on latest node and break on an older one, which is a semver-major change.

I have 300+ packages, and all of them run 200+ matrix jobs on every push and PR, because i test every supported minor - the minimum imo is testing every supported major.